nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,383 @@
+            }
+        } catch (Exception e) {
+            getLogger().error("Failed to load CDA document", e);
+            throw new ProcessException("Failed to load CDA document", e);
+        }
+        timingStats.add(String.format("Loaded document in %d ms", System.currentTimeMillis() - start));
+        return cd;
+    }
+
+    protected void loadMappings() {
+        long start = System.currentTimeMillis();
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        Properties mappings = new Properties();
+        try (InputStream is = classloader.getResourceAsStream("mapping.properties")){
+            mappings.load(is);
+            // each child element is key#value and multiple elements are separated by @
+            for (String property : mappings.stringPropertyNames()) {
+                String[] variables = StringUtils.split(mappings.getProperty(property), FIELD_SEPARATOR);
+                Map<String, String> map = new LinkedHashMap<String, String>();
+                for (String variable : variables) {
+                    String[] keyvalue = StringUtils.split(variable, KEY_VALUE_SEPARATOR);
+                    map.put(keyvalue[0], keyvalue[1]);
+                }
+                processMap.put(property, map);
+            }
+
+        } catch (IOException e) {
+            getLogger().error("Failed to load mappings", e);
+            throw new ProcessException("Failed to load mappings", e);
+        }
+
+        timingStats.add(String.format("Loaded mappings in %d ms", System.currentTimeMillis() - start));
+    }
+
+    private void logDetails(String json) {
+
+        if (getLogger().isTraceEnabled()) {
+            for (String timing : timingStats) {
+                getLogger().trace(timing);
+            }
+        }
+
+        if (getLogger().isDebugEnabled()) {
+            for (Entry<String, Map<String, String>> entry : processMap.entrySet()) {
+                getLogger().debug(String.format("Mapping: %s=%s", entry.getKey(), entry.getValue()));
+            }
+            for (Entry<String, String> entry : attributes.entrySet()) {
+                getLogger().debug(String.format("Attribute: %s=%s", entry.getKey(), entry.getValue()));
+            }
+            getLogger().debug(json);
+        }
+    }
+
+    protected class CDAValidationHandler implements ValidationHandler {
+        @Override
+        public void handleError(Diagnostic diagnostic) {
+            getLogger().error(new StringBuilder("ERROR: ").append(diagnostic.getMessage()).toString());
+        }
+
+        @Override
+        public void handleWarning(Diagnostic diagnostic) {
+            getLogger().warn(new StringBuilder("WARNING: ").append(diagnostic.getMessage()).toString());
+        }
+
+        @Override
+        public void handleInfo(Diagnostic diagnostic) {
+            getLogger().info(new StringBuilder("INFO: ").append(diagnostic.getMessage()).toString());
+        }
+    }
+
+    @Override

```

Comment:
Since you're not doing anything different for equals() and hashCode() you can just remove these.